specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -5230,20 +5230,23 @@
         "keyVaultKeyVersion",
         "keyVaultUri"
       ],
-      "description": "The configuration to use encryption with customer-managed keys. This information will be used to encrypt or decrypt the data you have protected using your own keys."
+      "description": "A fully-specified, user-defined encryption key in Azure Key Vault. Keys that you create and manage can be used to encrypt or decrypt data-at-rest in Azure Search, such as indexes and synonym maps."
     },
     "AzureActiveDirectoryApplicationCredentials": {
       "properties": {
         "applicationId": {
           "type": "string",
-          "description": "The is the ID of the Azure Active Directory Application that will be used to authenticate to your KeyVault. The Application ID is not to be confused with the Object ID of your Azure Active Directory Application."
+          "description": "The Application ID of an Azure Search service, when registered as an Azure Active Directory application, for the purpose of serving as a security principle for authentication to Azure Key Vault. The Application ID should not be confused with the Object ID for your application. Both IDs are GUIDs but have different purposes. The Application ID uniquely identifies the app on protocol transactions."
```

Comment:
principle -> principal

Reply:
I'd like to walk back some of the proposed text on this.   The doc that backs up the application ID concept is "Create a service principle".  I tried to work it in, but it doesn't clarify anything.  Nati's doc has a link so we should let that doc do the heavy lifting.

This version is the original with a few minor changes:

"The Application ID of an Azure Active Directory application requesting authentication to Azure Key Vault. The Application ID should not be confused with the Object ID, which looks similar."

Reply:
@shmed , @brjohnstmsft, @HeidiSteen - all of these descriptions seems too verbose, and seems to mix explaining what key-vault and AAD application are instead of how\why we use them. how about something on these lines:
**keyVaultKeyName** - The name of your Azure Key Vault key to be used to encrypt your data at rest
**keyVaultKeyVersion** - The version of your Azure Key Vault key to be used to encrypt your data at rest
**keyVaultUri** - The URI of your Azure Key Vault, also referred to as DNS name, that contains the key to be used to encrypt your index data at rest. An example URI might be https://my-keyvault-name.vault.azure.net

**accessCredentials** –  Optional Azure Active Directory credentials used for accessing your Azure Key Vault. Not required if using managed identity instead.",
    **applicationId** - An AAD Application ID that was granted the required access permissions to the Azure Key Vault that is to be used when encrypting your data at rest
    **applicationSecret** - The authentication key of the specified AAD application

Reply:
@natinimni I agree with your proposed changes to these specific property descriptions, except for the word "index" in your proposed text for *keyVaultUri* since these descriptions apply to synonym maps as well. Your descriptions are more concise, cutting out extraneous detail. I'm happy with the more verbose description we've landed on for `encryptionKey` though.